lib/manageiq_performance/middlewares/memory.rb

```diff
@@ -0,0 +1,55 @@
+module ManageIQPerformance
+  module Middlewares
+    module Memory
+      private
+
+      def memory_initialize
+      end
+
+      def memory_start(env)
+        Memory.memory_query_data = memory_parse_data
+      end
+
+      def memory_finish(env)
+        Memory.memory_query_data = memory_parse_data(Memory.memory_query_data)
+        save_report env, :memory,
+                    memory_short_form_data,
+                    memory_long_form_data
+      ensure
+        Memory.memory_query_data = nil
+      end
+
+      def memory_long_form_data
+        proc { Memory.memory_query_data }
+      end
+
+      alias_method memory_long_form_data memory_short_form_data
```

Comment:
I want `memory_long_form_data` to fetch the data.

I thought this was necessary for that method to call across.
I'll change and see how it goes